src/reorder.c

```diff
@@ -13,7 +13,7 @@ SEXP reorder(SEXP x, SEXP order)
     ncol = length(x);
     for (int i=0; i<ncol; i++) {
       SEXP v = VECTOR_ELT(x,i);
-      if (SIZEOF(v)!=4 && SIZEOF(v)!=8 && SIZEOF(v)!=16)
+      if (SIZEOF(v)!=1 && SIZEOF(v)!=4 && SIZEOF(v)!=8 && SIZEOF(v)!=16)
```

Comment:
I might do this test last since it's the least common

Reply:
I also had this in mind but was not sure if this potential minor speedup was worth loosing some kind of cleanliness of the kind. Same goes for checking `SIZEOF(x)` and ofc adding `raw` not as first if but as else branch below.